### PR TITLE
feature SC-3882 Content-Security-Policy code improved and added more rules to use.

### PR DESCRIPTION
**Pull request by dk-ciphron**
# Description

## Links to Tickets or other pull requests

- [https://ticketsystem.schul-cloud.org/browse/SC-3882](https://ticketsystem.schul-cloud.org/browse/SC-3882)

## Changes

- content security policy improved
- code clean up for csp
- required for better security

## Approval for review
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 46
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5523984&bid=18099470)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 46
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5523984&bid=18099470)

File: config/http-headers.js [hunk 1]
```diff
@@ -1,57 +1,86 @@
...
 	// Settings for HTTP Content-Security-Policy Header
 	contentSecurityPolicy: {
 		// Default Content-Security-Policy Header for every site
-		corsDefault: {
-			defaultSrc: "data: blob: 'self' 'unsafe-inline' https://scchat.schul-cloud.org https://storage.schul-cloud.org https://libreoffice.schul-cloud.org https://docs.schul-cloud.org https://etherpad.schul-cloud.org https://blog.schul-cloud.org https://sc-content-resources.schul-cloud.org",
-			scriptSrc: "data: blob: 'self' 'unsafe-inline'",
-			objectSrc: '',
+		// Use 'strict-dynamic' 'nonce-<nonceValue>' (nonceValue auto generated) to create a whitelist
+	    corsDefault: {
+	    	defaultSrc: "'self' data: blob: https://api.schul-cloud.org https://scchat.schul-cloud.org https://s3.hidrive.strato.com https://libreoffice.schul-cloud.org https://docs.schul-cloud.org https://edtrio.schul-cloud.org https://etherpad.schul-cloud.org https://blog.schul-cloud.org https://sc-content-resources.schul-cloud.org https://sentry.schul-cloud.dev https://open.hpi.de https://upload.wikimedia.org https://user-images.githubusercontent.com",
```
Comment (CeEv) [inline config/http-headers.js:15]: https://api.schul-cloud.org ist nur eine unserer Instanzen. Da muss noch mal jemand drauf schauen wie bisher die unterschiedlichen Instanzen gehandhabt wurden. Das trifft auch auf https://edtrio.schul-cloud.org zu.

File: config/http-headers.js [hunk 2]
```diff
@@ -65,14 +94,16 @@ const config = {
 	},
 	// Additional default Security header can be set - key reprensents the HTTP header and the value the value of the header
 	additionalSecurityHeader: {
-		'Strict-Transport-Security': 'max-age=63072000; includeSubDomains; preload',
 		'X-Frame-Options': 'sameorigin',
+		'X-Robots-Tag': 'none',
```
Comment (CeEv) [inline config/http-headers.js:98]: Dafür hatten wir auch noch ein extra Ticket oder?

File: app.js [hunk 3]
```diff
@@ -66,11 +67,19 @@ if (KEEP_ALIVE) {
 	});
 }
 
+// disable x-powered-by header
+app.disable('x-powered-by');
```
Comment (CeEv) [inline app.js:71]: Ist Teil eines anderen PRs, könnte man dann gleich mit auf review stellen.

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 11
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5523984&bid=18099470)

**Comment by malcm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 11
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5523984&bid=18099470)

**Pull request merged**